docs/components/rule.md

```diff
@@ -1,36 +1,139 @@
+query reliability. For Prometheus it is unlikely to have alert/recording rule evaluation failure as evaluation is local.
+
+For Ruler the read path is distributed, since most likely ruler is querying Thanos Querier which gets data from remote Store APIs. 
+
+This means that **query failure** are more likely to happen, that's why clear strategy on what will happen to alert and during query
+unavailability is the key.
+
+## Partial Response
+
+See [this](query.md#PartialResponse) on initial info.
+
+Rule allows to specify rule groups with additional field that controls PartialResponseStrategy e.g:
 
 ```yaml
-type: GCS
-config:
-  bucket: example-bucket
+groups:
+- name: "warn strategy"
+  partial_response_strategy: "warn"
+  rules:
+  - alert: "some"
+    expr: "up"
+- name: "abort strategy"
+  partial_response_strategy: "abort"
+  rules:
+  - alert: "some"
+    expr: "up"
+- name: "by default strategy is abort"
+  rules:
+  - alert: "some"
+    expr: "up"
 ```
 
+It is recommended to keep partial response to `abort` for alerts and that is the default as well.
+
+Essentially for alerting having partial response can result in symptom being missed by Rule's alert.
+
+## Must have: essential Ruler alerts! 
+
+To be sure that alerting works it is essential to monitor Ruler and alert from another **Scraper (Prometheus + sidecar)** that sits in same cluster.
+
+The most important metrics to alert on are:
+
+* `thanos_alert_sender_alerts_dropped_total`. If greater than 0 it means that rule triggered alerts are not being sent to alertmanager which might
+indicate connection, incompatibility or misconfiguration problems.
+
+* `prometheus_rule_evaluation_failures_total`. If greater than 0 it means that rule failed to be evaluated which results in
+either gap in rule or potentially ignored alert. Alert heavily on this if this happens for longer than your alert thresholds.
+`strategy` label will tell you if failures comes from rules that tolerates [partial response](rule.md#PartialResponse) or not.
+
+* `prometheus_rule_group_last_duration_seconds < prometheus_rule_group_interval_seconds`  If the difference is heavy it means 
+that rule evaluation took more time than scheduled interval. It can indicate your query backend (e.g Querier) takes too much time 
+to evaluate the query, that is not fast enough to fill the rule. This might indicate other problems like slow StoreAPis or 
+too complex query expression in rule. 
+
+* `thanos_rule_evaluation_with_warnings_total`. If you choose to use Rules and Alerts with [partial response strategy](rule.md#PartialResponse)
+equals "warn", this metric will tell you how many evaluation ended up with some kind of warning. To see the actual warnings
+see WARN log level. This might suggest that those evaluations returns partial response and might be or not accurate.
+
+Those metrics are important for vanilla Prometheus as well, but even more important when we rely on (sometimes WAN) network.
+
```

Comment:
Maybe here we should also add about the new timeout in `store.response-timeout`, i.e. if you have a slow Store API it will time them out and you will see partial response or log errors in the abort case?